MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -2382,6 +2387,145 @@ - (void)didMXRoomInitialSynced:(NSNotification *)notif
+                NSMutableArray<MXEvent*> *removedEvents;
+                NSUInteger remainingEvents = [bubbleData removeEventsFromEvent:event.eventId removedEvents:&removedEvents];
+
+                // If there is no more events in this bubble, remove it
+                if (0 == remainingEvents)
+                {
+                    eventIsFirstInBubble = YES;
+                    @synchronized (eventsToProcessSnapshot)
+                    {
+                        [bubbles removeObjectAtIndex:bubbleDataIndex];
+                        bubbleDataIndex--;
+                    }
+                }
+
+                // Create a dedicated bubble for the attachment
+                if (removedEvents.count)
+                {
+                    Class class = [self cellDataClassForCellIdentifier:kMXKRoomBubbleCellDataIdentifier];
+
+                    id<MXKRoomBubbleCellDataStoring> newBubbleData = [[class alloc] initWithEvent:removedEvents[0] andRoomState:self.room.state andRoomDataSource:self];
+
+                    if (eventIsFirstInBubble)
+                    {
+                        // Apply same config as before
+                        newBubbleData.isPaginationFirstBubble = bubbleData.isPaginationFirstBubble;
+                        newBubbleData.shouldHideSenderInformation = bubbleData.shouldHideSenderInformation;
+                    }
+                    else
+                    {
+                        // This new bubble is not the first. Show nothing
+                        newBubbleData.isPaginationFirstBubble = NO;
+                        newBubbleData.shouldHideSenderInformation = YES;
+                    }
+
+                    // Update bubbles mapping
+                    @synchronized (eventIdToBubbleMap)
+                    {
+                        eventIdToBubbleMap[event.eventId] = newBubbleData;
+                    }
+
+                    @synchronized (eventsToProcessSnapshot)
+                    {
+                        [bubbles insertObject:newBubbleData atIndex:bubbleDataIndex + 1];
+                    }
+                }
+
+                // And put other cutted events in another bubble
+                if (removedEvents.count > 1)
+                {
+                    Class class = [self cellDataClassForCellIdentifier:kMXKRoomBubbleCellDataIdentifier];
+
+                    id<MXKRoomBubbleCellDataStoring> newBubbleData;
+                    for (NSUInteger i = 1; i < removedEvents.count; i++)
+                    {
+                        MXEvent *removedEvent = removedEvents[i];
+                        if (i == 1)
+                        {
+                            newBubbleData = [[class alloc] initWithEvent:removedEvent andRoomState:self.room.state andRoomDataSource:self];
+                        }
+                        else
+                        {
+                            [newBubbleData addEvent:removedEvent andRoomState:self.room.state];
```

Comment:
Please check here [class instancesRespondToSelector:@selector(addEvent:andRoomState:)]

even if this is not possible that the call does not respond to this selector